### PR TITLE
[Feature][Dataobject] add confirm dialog to empty button of relations and add possibility to disable clear relations

### LINKED ISSUE #11375: [Improvement]: Object ManyToMany Relation Grid Controls
### Improvement description

The Object ManyToMany Relation Grid has two UX problems:

The "Delete" or "Clear" Button clears all data without asking a second time. And it can not be undone easily, because it saves the object already. This is very annoying if someone accidentally clicks this button with a lot of related objects.

The green "Add" button makes no sense too. How should someone use this function without knowing how to store the object in the correct folder structure. The "Add" button could be the search field too. The button illustrates that we want to add something to the table. But it creates an object.

It should be possible to decide if these buttons should be there or not. Now it is not possible to hide the delete and add buttons.

ManyToMany Controls

### PR DESCRIPTION
**Pull request by dpfahlbusch**
## Changes in this pull request  
Resolves #11375

### CODE REVIEW

**Comment by robertSt7** [comment]
@dpfahlbusch  It doesn't work for me, I think you need to call the empty function like: this.empty().bind(this)

Could you please also 
- [x] set the default value of allowToClearRelation to true
- [x] add the english translations 
- [x] add the allowToClearRelation option for every relation

Thank you

**Comment by dpfahlbusch** [comment]
@robertSt7 done :)

File: bundles/AdminBundle/public/js/pimcore/object/tags/advancedManyToManyRelation.js [hunk 1]
```diff
@@ -557,12 +557,19 @@ pimcore.object.tags.advancedManyToManyRelation = Class.create(pimcore.object.tag
...
-                {
+            if (this.fieldConfig.allowToClearRelation) {
+                toolbarItems.push({
                     xtype: "button",
                     iconCls: "pimcore_icon_delete",
-                    handler: this.empty.bind(this)
-                },
+                    handler: function () {
+                        pimcore.helpers.deleteConfirm(t('relations'), t('all'), function () {
+                            this.empty.bind(this);
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/object/tags/advancedManyToManyRelation.js]:
```suggestion
                            this.empty();
```

File: bundles/AdminBundle/public/js/pimcore/object/tags/advancedManyToManyRelation.js [hunk 2]
```diff
@@ -557,12 +557,19 @@ pimcore.object.tags.advancedManyToManyRelation = Class.create(pimcore.object.tag
...
-            toolbarItems = toolbarItems.concat([
-                {
+            if (this.fieldConfig.allowToClearRelation) {
+                toolbarItems.push({
                     xtype: "button",
                     iconCls: "pimcore_icon_delete",
-                    handler: this.empty.bind(this)
-                },
+                    handler: function () {
+                        pimcore.helpers.deleteConfirm(t('relations'), t('all'), function () {
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/object/tags/advancedManyToManyRelation.js]:
```suggestion
                        pimcore.helpers.deleteConfirm(t('all'), t('relations'), function () {
```

File: bundles/AdminBundle/public/js/pimcore/object/tags/manyToManyObjectRelation.js [hunk 3]
```diff
@@ -584,12 +584,18 @@ pimcore.object.tags.manyToManyObjectRelation = Class.create(pimcore.object.tags.
...
-            toolbarItems = toolbarItems.concat([
-                {
+            if (this.fieldConfig.allowToClearRelation) {
+                toolbarItems.push({
                     xtype: "button",
                     iconCls: "pimcore_icon_delete",
-                    handler: this.empty.bind(this)
-                },
+                    handler: function () {
+                        pimcore.helpers.deleteConfirm(t('relations'), t('all'), function () {
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/object/tags/manyToManyObjectRelation.js]:
```suggestion
                        pimcore.helpers.deleteConfirm(t('all'), t('relations'), function () {
```

File: bundles/AdminBundle/public/js/pimcore/object/tags/manyToManyRelation.js [hunk 4]
```diff
@@ -358,18 +358,25 @@ pimcore.object.tags.manyToManyRelation = Class.create(pimcore.object.tags.abstra
...
                 handler: this.openSearchEditor.bind(this)
             }
         ]);
 
+        if (this.fieldConfig.allowToClearRelation) {
+            toolbarItems.push({
+                xtype: "button",
+                iconCls: "pimcore_icon_delete",
+                handler: function () {
+                    pimcore.helpers.deleteConfirm(t('relations'), t('all'), function () {
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/object/tags/manyToManyRelation.js]:
```suggestion
                    pimcore.helpers.deleteConfirm(t('all'), t('relations'), function () {
```

**Review by robertSt7 — APPROVED** [review]
@dpfahlbusch Thanks :) 
I think we should display the "all" before the "Relations". What do you think?

File: bundles/AdminBundle/public/js/pimcore/object/classes/data/advancedManyToManyObjectRelation.js [hunk 5]
```diff
@@ -196,6 +196,13 @@ pimcore.object.classes.data.advancedManyToManyObjectRelation = Class.create(pimc
             value: this.datax.allowToCreateNewObject
         });
 
+        this.specificPanel.add({
+            xtype: "checkbox",
+            boxLabel: t("allow_to_clear_relation"),
+            name: "allowToClearRelation",
+            value: this.datax.allowToClearRelation
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/object/classes/data/advancedManyToManyObjectRelation.js]:
```suggestion
            value: this.datax.allowToClearRelation ?? true
```

File: bundles/AdminBundle/public/js/pimcore/object/classes/data/advancedManyToManyRelation.js [hunk 6]
```diff
@@ -376,6 +376,13 @@ pimcore.object.classes.data.advancedManyToManyRelation = Class.create(pimcore.ob
             value: this.datax.allowMultipleAssignments
         });
 
+        this.specificPanel.add({
+            xtype: "checkbox",
+            boxLabel: t("allow_to_clear_relation"),
+            name: "allowToClearRelation",
+            value: this.datax.allowToClearRelation
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/object/classes/data/advancedManyToManyRelation.js]:
```suggestion
            value: this.datax.allowToClearRelation ?? true
```

File: bundles/AdminBundle/public/js/pimcore/object/classes/data/manyToManyObjectRelation.js [hunk 7]
```diff
@@ -203,6 +203,12 @@ pimcore.object.classes.data.manyToManyObjectRelation = Class.create(pimcore.obje
                 name: "allowToCreateNewObject",
                 value: this.datax.allowToCreateNewObject
             });
+            this.specificPanel.add({
+                xtype: "checkbox",
+                boxLabel: t("allow_to_clear_relation"),
+                name: "allowToClearRelation",
+                value: this.datax.allowToClearRelation
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/object/classes/data/manyToManyObjectRelation.js]:
```suggestion
                value: this.datax.allowToClearRelation ?? true
```

File: bundles/AdminBundle/public/js/pimcore/object/classes/data/manyToManyRelation.js [hunk 8]
```diff
@@ -193,6 +193,13 @@ pimcore.object.classes.data.manyToManyRelation = Class.create(pimcore.object.cla
                         name: 'pathFormatterClass',
                         value: this.datax.pathFormatterClass
                     },
+                    {
+                        xtype: "checkbox",
+                        width: 600,
+                        fieldLabel: t("allow_to_clear_relation"),
+                        name: "allowToClearRelation",
+                        value: this.datax.allowToClearRelation
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/object/classes/data/manyToManyRelation.js]:
```suggestion
                        value: this.datax.allowToClearRelation ?? true
```

File: bundles/AdminBundle/public/js/pimcore/object/classes/data/manyToOneRelation.js [hunk 9]
```diff
@@ -170,6 +170,11 @@ pimcore.object.classes.data.manyToOneRelation = Class.create(pimcore.object.clas
                         fieldLabel: t("path_formatter_service"),
                         name: 'pathFormatterClass',
                         value: this.datax.pathFormatterClass
+                    }, {
+                        xtype: "checkbox",
+                        boxLabel: t("allow_to_clear_relation"),
+                        name: "allowToClearRelation",
+                        value: this.datax.allowToClearRelation
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/object/classes/data/manyToOneRelation.js]:
```suggestion
                        value: this.datax.allowToClearRelation ?? true
```

**Review by robertSt7 — APPROVED** [review]
@dpfahlbusch One more thing. Could you also please add some documentation like: [Relations]: Add confirm dialog to empty button of relations and add possibility to disable clear relations in the class layout

**Comment by robertSt7** [comment]
@dpfahlbusch Thank you very much

**Pull request merged**